pkg/crd/flatten_all_of_test.go

```diff
@@ -241,6 +241,25 @@ var _ = Describe("AllOf Flattening", func() {
 				},
 			}))
 		})
+
+		It("should merge XValidation fields", func() {
```

Comment:
Thx for the unit test, let's please also add test coverage in https://github.com/kubernetes-sigs/controller-tools/blob/5e5bc88a01cab8ebff8c9fc4d1e90bb858139186/pkg/crd/testdata/cronjob_types.go

Reply:
Done! Added it on https://github.com/kubernetes-sigs/controller-tools/pull/998/commits/43611064a1be33cfab6d25aa1b03bd11a1b39d3c